examples/arkode/C_serial/lsrk_analytic.c

```diff
@@ -0,0 +1,317 @@
+  printf("   abstol = %.1" ESYM "\n\n", abstol);
+
+  /* Initialize data structures */
+  y = N_VNew_Serial(NEQ, ctx); /* Create serial vector for solution */
+  if (check_flag((void*)y, "N_VNew_Serial", 0)) { return 1; }
+  NV_Ith_S(y, 0) = SUN_RCONST(0.0); /* Specify initial condition */
+
+  /* Call LSRKStepCreate to initialize the ARK timestepper module and
+     specify the right-hand side function in y'=f(t,y), the initial time
+     T0, and the initial dependent variable vector y. */
+  arkode_mem = LSRKStepCreate(f, NULL, T0, y, ctx);
+  if (check_flag((void*)arkode_mem, "ARKStepCreate", 0)) { return 1; }
+
+  /* Set routines */
+  flag = ARKodeSetUserData(arkode_mem,
+                           (void*)&lambda); /* Pass lambda to user functions */
+  if (check_flag(&flag, "ARKodeSetUserData", 1)) { return 1; }
+
+  /* Specify tolerances */
+  flag = ARKodeSStolerances(arkode_mem, reltol, abstol);
+  if (check_flag(&flag, "ARKStepSStolerances", 1)) { return 1; }
+
+  /* Specify user provided spectral radius */
+  flag = LSRKStepSetDomEigFn(arkode_mem, dom_eig);
+  if (check_flag(&flag, "LSRKStepSetDomEigFn", 1)) { return 1; }
+
+  /* Specify after how many successful steps dom_eig is recomputed 
+     Note that nsteps = 0 refers to constant dominant eigenvalue */
+  flag = LSRKStepSetDomEigFrequency(arkode_mem, 0);
+  if (check_flag(&flag, "LSRKStepSetDomEigFrequency", 1)) { return 1; }
+
+  /* Specify max number of stages allowed */
+  flag = LSRKStepSetMaxNumStages(arkode_mem, 200);
+  if (check_flag(&flag, "LSRKStepSetMaxNumStages", 1)) { return 1; }
+
+  /* Specify max number of steps allowed */
+  flag = ARKodeSetMaxNumSteps(arkode_mem, 1000);
+  if (check_flag(&flag, "ARKodeSetMaxNumSteps", 1)) { return 1; }
+
+  /* Specify safety factor for user provided dom_eig */
+  flag = LSRKStepSetDomEigSafetyFactor(arkode_mem, 1.01);
+  if (check_flag(&flag, "LSRKStepSetDomEigSafetyFactor", 1)) { return 1; }
+
+  /* Specify the Runge--Kutta--Legendre LSRK method */
+  flag = LSRKStepSetMethod(arkode_mem, ARKODE_LSRK_RKL_2);
+  if (check_flag(&flag, "LSRKStepSetMethod", 1)) { return 1; }
+
+  /* Open output stream for results, output comment line */
+  UFID = fopen("solution.txt", "w");
+  fprintf(UFID, "# t u\n");
+
+  /* output initial condition to disk */
+  fprintf(UFID, " %.16" ESYM " %.16" ESYM "\n", T0, NV_Ith_S(y, 0));
+
+  /* Main time-stepping loop: calls ARKodeEvolve to perform the integration, then
+     prints results.  Stops when the final time has been reached */
+  t    = T0;
+  tout = T0 + dTout;
+  printf("        t           u\n");
+  printf("   ---------------------\n");
+  while (Tf - t > 1.0e-15)
+  {
+    flag = ARKodeEvolve(arkode_mem, tout, y, &t, ARK_NORMAL); /* call integrator */
+    if (check_flag(&flag, "ARKodeEvolve", 1)) { break; }
+    printf("  %10.6" FSYM "  %10.6" FSYM "\n", t,
+           NV_Ith_S(y, 0)); /* access/print solution */
+    fprintf(UFID, " %.16" ESYM " %.16" ESYM "\n", t, NV_Ith_S(y, 0));
```

Comment:
Accessor macros like `NV_Ith_S` are generally considered deprecated and should not be used in new examples. Instead use `N_VGetArrayPointer` to access the data. 
```suggestion
    printf("  %10.6" FSYM "  %10.6" FSYM "\n", t,
           N_VGetArrayPointer(y)[0]); /* access/print solution */
    fprintf(UFID, " %.16" ESYM " %.16" ESYM "\n", t, N_VGetArrayPointer(y)[0]);
```